pmd-core/src/main/java/net/sourceforge/pmd/PMDConfiguration.java

```diff
@@ -448,75 +469,162 @@ public List<String> getAllInputPaths() {
+        Objects.requireNonNull(inputPath);
+        this.inputPaths.add(Paths.get(inputPath));
+    }
+
+    /**
+     * Add an input path. It is not split on commas.
+     *
+     * @throws NullPointerException If the parameter is null
+     */
+    public void addInputPath(Path inputPath) {
         Objects.requireNonNull(inputPath);
         this.inputPaths.add(inputPath);
     }
 
+    /**
+     * @deprecated Use {@link #getInputFile()}
+     */
+    @Deprecated
     public String getInputFilePath() {
+        return inputFilePath == null ? null : inputFilePath.toString();
+    }
+
+    public Path getInputFile() {
         return inputFilePath;
     }
 
+    /**
+     * @deprecated Use {@link #getIgnoreFile()}
+     */
+    @Deprecated
     public String getIgnoreFilePath() {
+        return ignoreFilePath == null ? null : ignoreFilePath.toString();
+    }
+
+    public Path getIgnoreFile() {
         return ignoreFilePath;
     }
 
     /**
      * The input file path points to a single file, which contains a
      * comma-separated list of source file names to process.
      *
-     * @param inputFilePath
-     *            path to the file
+     * @param inputFilePath path to the file
+     * @deprecated Use {@link #setInputFilePath(Path)}
      */
+    @Deprecated
     public void setInputFilePath(String inputFilePath) {
+        this.inputFilePath = inputFilePath == null ? null : Paths.get(inputFilePath);
+    }
+
+    /**
+     * The input file path points to a single file, which contains a
+     * comma-separated list of source file names to process.
+     *
+     * @param inputFilePath path to the file
+     */
+    public void setInputFilePath(Path inputFilePath) {
         this.inputFilePath = inputFilePath;
     }
 
     /**
      * The input file path points to a single file, which contains a
      * comma-separated list of source file names to ignore.
      *
-     * @param ignoreFilePath
-     *            path to the file
+     * @param ignoreFilePath path to the file
+     * @deprecated Use {@link #setIgnoreFilePath(Path)}
      */
+    @Deprecated
     public void setIgnoreFilePath(String ignoreFilePath) {
+        this.ignoreFilePath = ignoreFilePath == null ? null : Paths.get(ignoreFilePath);
+    }
+
+    /**
+     * The input file path points to a single file, which contains a
+     * comma-separated list of source file names to ignore.
+     *
+     * @param ignoreFilePath  path to the file
+     */
+    public void setIgnoreFilePath(Path ignoreFilePath) {
         this.ignoreFilePath = ignoreFilePath;
     }
 
     /**
      * Get the input URI to process for source code objects.
      *
      * @return URI
+     * @deprecated Use {@link #getUri}
      */
+    @Deprecated
     public String getInputUri() {
+        return inputUri == null ? null : inputUri.toString();
+    }
+
+    /**
+     * Get the input URI to process for source code objects.
+     *
+     * @return URI
+     */
+    public URI getUri() {
```

Comment:
I think this name is too vague and we should at least rename it (eg to `getSourceDbUri` or so)... Maybe it's also the right time to deprecate this feature altogether. It's untested, and it's completely unclear how to use it at all.

Reply:
This is/was actually a PMD 7 misc task (in #2524). I've created a separate issue for this for pmd 7 -> #4216 

Since this feature exists already, we should document how to use it in any case.
Remaining question is: Should we remove it with pmd 7?